src/ape/pytest/runners.py

```diff
@@ -153,14 +156,20 @@ def pytest_collection_finish(self, session):
             self._provider_context.push_provider()
             self._provider_is_connected = True
 
-    def pytest_sessionfinish(self):
+    def pytest_terminal_summary(self, terminalreporter):
         """
-        Called after whole test run finished, right before returning the exit
-        status to the system.
-
-        **NOTE**: This hook fires even when exceptions occur, so we cannot
-        assume the provider successfully connected.
+        Add a section to terminal summary reporting.
+        When ``--gas`` is active, outputs the gas profile report.
         """
+        if self.pytest_config.getoption("--gas"):
+            terminalreporter.section("Gas Profile")
+            gas_report = self.receipt_capture.gas_report
+            if gas_report:
+                tables = parse_gas_table(gas_report)
+                rich_print(*tables)
+            else:
+                terminalreporter.write_line(f"{LogLevel.WARNING.name}: No gas usage data found.")
+
         if self._provider_is_connected:
```

Comment:
Do we want this under `pytest_terminal_summary` too?

Reply:
It at least needs to happen _after_ outputting the tables, and before it wasn't. There is probably a better a hook for it though.

Reply:
Well, my worry is if there is somehow a flag to skip this hook from operating, it might break the cleanup code

Reply:
Resolved in 3d36109

Reply:
Nice!